README.md

```diff
@@ -1,277 +1,1037 @@
+Sarah looks up all swap addresses willing to swap ADA/HOSKY. Sarah finds Mike's address.
+Sarah gives Mike 10 ada and receives 10 hosky.
+Sarah gives Charlie 10 hosky and receives 10 agix.
+Sarah gives Bob 5 agix and receives 10 dust.
+Sarah gives Alice 10 dust and receives 10 ada.
+Sarah now has her original 10 ada plus an additional 5 agix.
+This all occurs in one transaction where Sarah pays the transaction fee.
+```
 
-> **Note**
-> 
-> [Present and future discussions with the community](https://github.com/cardano-foundation/CIPs/pull/418) will help determine if this feature is indeed warranted.
+On net, Sarah pays the transaction fee and receives 5 agix in return, while four swaps are
+fulfilled. As shown in this example, Sarah fulfills *both* the AGIX/DUST swap and the HOSKY/AGIX
+swap by "passing through" those pairs on her way back to ada. As long as the entry and exit pairs
+(in this case ADA/HOSKY and DUST/ADA) have enough liquidity, arbitragers can spread that liquidity
+into less liquid swap pairs. And since two-way swaps naturally incentivize providing liquidity for
+major trading pairs, this requirement for entry and exit liquidity is naturally incentivized to be
+satisfied.
+
+Since the current design is capable of handling 11-14 swaps in a single transaction, there is ample
+flexibility for finding arbitrage opportunities. For example, maybe the 3rd swap is at a slight loss
+but the 5th swap is such a good arbitrage that it more than makes up for it resulting in a net
+profit for the arbitrage.
+
+##### The Stablecoin <-> Stablecoin Two-Way Swap
+
+Currently, DApps treat stablecoins individually (ie, you may be able to borrow a loan in DJED but
+not USDC). This means, even though all stablecoins are practically the US dollar, the liquidity is
+fractured across all stablecoins. Two-way swaps change this.
+
+Imagine if Charlie has a swap for DJED <-> USDC where converting assets in either direction requires
+paying a 1% premium. The very existence of this swap, combined with the fact that Cardano-Swaps is
+freely composable with all DApps, means users can freely convert between stablecoins as needed. The
+liquidity would no longer be fractured. Alice could convert her DJED to USDC to buy an options
+contract that is asking for USDC. She just has to pay Charlie the 1% fee for the conversion.
+
+Since two-way swaps can go in either direction, these swaps do not need to be reset after each swap.
+For example, after Alice deposits DJED into the swap to claim USDC, Bob can now use the swap to
+convert his USDC to DJED by paying a 1% fee. Either way, Charlie makes 1% per swap and *is paid
+directly in the stablecoin being deposited*. And since the stablecoins are supposed to be 1:1 with
+the US dollar, the only risk is a peg-break. Anyone can be a liquidity provider and profit like
+Charlie does in this example.
+
+Being able to treat all stablecoins as a single stablecoin will be a game changer for DeFi.
```

Comment:
I am not sure if I should add something after this line. AFAICT this ability to unify stablecoin liquidity is unique to Cardano-Swaps *because* it is p2p and natively composable. I don't think either Axo or GeniusYield can do this since their batchers would need to execute the transaction for you. It is the same as the "batchers can't trustlessly compose arbitrary DApps" argument. They may support some compositions but not others which would mean some stablecoins conversions are still not possible. This asymmetry of support for conversion may be worse than not supporting conversions at all; it effectively means Axo and GeniusYield can choose winners and loses among stablecoins.